cartographer_ros/cartographer_ros/map_options.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_ROS_MAP_OPTIONS_H_
+#define CARTOGRAPHER_ROS_MAP_OPTIONS_H_
+
+#include <string>
+
+#include "cartographer/common/lua_parameter_dictionary.h"
+#include "cartographer/common/port.h"
+#include "cartographer/mapping/map_builder.h"
+#include "cartographer_ros/sensor_bridge.h"
+
+namespace cartographer_ros {
+
+struct MapOptions {
```

Comment:
Now I am working on making "multibot_node_main" which needs this struct. I am thinking when the node boots up, it will make map_builder with this map options, and a ros service of starting trajectory will send a trajectory_options and start a new trajectory.